Slim/Schema/Genre.pm

```diff
@@ -35,6 +41,40 @@ use Slim::Utils::Log;
 	$class->resultset_class('Slim::Schema::ResultSet::Genre');
 }
 
+sub loadMyClassicalGenreMap {
+	my $prefs = preferences('server');
+	%$myClassicalGenreMap = map {$_ => 1} split(/\s*,\s*/, uc($prefs->get('myClassicalGenres')));
+	if ( !%$myClassicalGenreMap ) {
+		$myClassicalGenreIds = undef;
+		return;
+	} else {
+		# also load genre ids from database
+		my @genreNames = keys %$myClassicalGenreMap;
+		my $dbh = Slim::Schema->dbh;
+		my $sql = 'SELECT GROUP_CONCAT(id) FROM genres WHERE UPPER(name) IN (' . join(', ', map {'?'} @genreNames) . ')';
+		my $sth = $dbh->prepare_cached($sql);
+		$sth->execute(@genreNames);
+		($myClassicalGenreIds) = $sth->fetchrow_array;
+		$sth->finish;
+	}
+}
+
+sub isMyClassicalGenre {
+	loadMyClassicalGenreMap() if !$myClassicalGenreMap;
+	my $class = shift;
+	my $genres = shift;
+	my $sep = shift;
```

Comment:
Not to be fixed now. But for next time 😉: I prefer the params reading to be the first thing after the `sub` declaration. And them to be read in list style when dealing with multiple params:

```perl
my ($class, $genres, $sep) = @_;
```

It's just less chatty.